e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh

```diff
@@ -178,17 +178,19 @@ az keyvault secret set --vault-name "$kv_name" --name "applicationInsightsConnec
 # ###########################
 echo "validate_password function..."
 # ###########################
-###if there is an hyphen in the first position replace by R
+###if there is an hyphen reset
 # ###########################
 validate_password() {
     local password="$1"
+    local sprincipal="$2"
    
     if [[ -z "$password" || "${password:0:1}" == "-" ]]; then
-        password="R${password:1}"
-        echo "$password"
-     
+       local sp_stor_outval
+       sp_stor_outval=$(az ad sp credential reset --id "$sprincipal" )
+       password=$(echo "$sp_stor_outval" | jq -r '.password')
```

Comment:
It is unlikely, BUT what happens if the reset password starts with "-" as well? Can we recursively call the validate password on the new generated password?

Reply:
Yeah... true... But I thought 2X would be unlikely.. that is why I did not added the recursive, so I did not want to make it more complicate... even once it is quite  random right>